source/includes/fundamentals/code-snippets/CompoundOperatorsIndividualExamples.java

```diff
@@ -0,0 +1,92 @@
+package com.mycompany.app;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.*;
+import com.mongodb.client.result.InsertOneResult;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+
+import java.util.ArrayList;
+import java.util.concurrent.TimeUnit;
+
+public class CompoundOperatorsIndividualExamples {
+
+    private static final String COLLECTION = "compound-ops";
+    private static final String DATABASE = "test";
+
+    public static void main(String[] args) throws InterruptedException {
+        CompoundOperatorsIndividualExamples.resetExample();
+        CompoundOperatorsIndividualExamples examples = new CompoundOperatorsIndividualExamples();
+        examples.findOneAndUpdateExample();
+        examples.findOneAndReplaceExample();
+        examples.findOneAndDeleteExample();
+    }
+
+    public static void resetExample() {
+        MongoCollection<Document> collection = getCollection();
+        collection.deleteMany(new Document());
+        Document insert_pizza = new Document("_id", 1).append("food", "donut").append("color", "green");
+        Document insert_pear = new Document("_id", 2).append("food", "pear").append("color", "yellow");
+        ArrayList<Document> docs = new ArrayList<Document>();
+        docs.add(insert_pizza);
+        docs.add(insert_pear);
+        collection.insertMany(docs);
+    }
+
+    public static MongoCollection<Document> getCollection() {
+        String uri = System.getenv("DRIVER_URL");
+        MongoClient mongoClient = MongoClients.create(uri);
+        MongoDatabase database = mongoClient.getDatabase(DATABASE);
+        MongoCollection<Document> collection = database.getCollection(COLLECTION);
+        return collection;
+    }
+
+    private void findOneAndUpdateExample() {
+        System.out.println("Starting find one and update example...");
+        MongoCollection<Document> collection = getCollection();
+        //start findOneAndUpdate-example
+        // <MongoCollection set up code here>
+        Bson projection = Projections.excludeId();
+        Bson filter = Filters.eq("color", "green");
+        Bson update = Updates.set("food", "pizza");
+        FindOneAndUpdateOptions options = new FindOneAndUpdateOptions().
```

Comment:
I tried following this as part of my PR to use ``findOneAndUpdate``, but I had trouble retrieving the document after the update. By default, this returns the state of the document before the update. Similar to your ``findOneAndReplace`` example, you need to specify the ``returnDocument()``. (This may be the case for ``findOneAndDelete`` too)

```suggestion
        FindOneAndUpdateOptions options = new FindOneAndUpdateOptions().
        returnDocument(ReturnDocument.AFTER).
```

Reply:
This comment was discussed synchronously. We agreed this should not be updated to `returnDocumnet(ReturnDocument.AFTER)` as this may imply that the default behavior of the driver is undesirable if all examples do not use it. The solution arrived at was to add a comment above this line of code clarifying that this will return the document your update operation.